src/dsp_tools/commands/excel2json/resources.py

```diff
@@ -197,47 +191,50 @@ def excel2resources(
-    if any(all_classes_df.get(lang) is not None for lang in languages):
-        warnings.warn(
-            f"The file {excelfile} uses {languages} as column titles, which is deprecated. "
-            f"Please use {[f'label_{lang}' for lang in languages]}"
-        )
+    _validate_excel_file(classes_df, df_dict)
 
     # transform every row into a resource
-    resources = [_row2resource(row, excelfile) for i, row in all_classes_df.iterrows()]
-
-    # check if resource names are unique
-    all_names = [r["name"] for r in resources]
-    if duplicates := {
-        index + 2: resdef["name"] for index, resdef in enumerate(resources) if all_names.count(resdef["name"]) > 1
-    }:
-        err_msg = (
-            f"Resource names must be unique inside every ontology, "
-            f"but your Excel file '{excelfile}' contains duplicates:\n"
-        )
-        for row_no, resname in duplicates.items():
-            err_msg += f" - Row {row_no}: {resname}\n"
-        raise UserError(err_msg)
+    resources = [_row2resource(row, df_dict[row["name"]]) for i, row in classes_df.iterrows()]
 
     # write final "resources" section into a JSON file
-    _validate_resources(resources_list=resources, excelfile=excelfile)
+    _validate_resources(resources_list=resources)
 
     if path_to_output_file:
         with open(file=path_to_output_file, mode="w", encoding="utf-8") as file:
             json.dump(resources, file, indent=4, ensure_ascii=False)
             print(f"resources section was created successfully and written to file '{path_to_output_file}'")
 
     return resources, True
+
+
+def _validate_excel_file(classes_df: pd.DataFrame, df_dict: dict[str, pd.DataFrame]) -> None:
+    for index, row in classes_df.iterrows():
+        index = int(str(index))  # index is a label/index/hashable, but we need an int
+        if not check_notna(row["super"]):
+            raise UserError(
+                f"Sheet 'classes' of 'resources.xlsx' has a missing value in row {index + 2}, column 'super'"
+            )
+    if any(classes_df.get(lang) is not None for lang in languages):
+        warnings.warn(
+            f"The file 'resources.xlsx' uses {languages} as column titles, which is deprecated. "
+            f"Please use {[f'label_{lang}' for lang in languages]}"
+        )
+    duplicate_check = check_column_for_duplicate(classes_df, "name")
+    if duplicate_check:
+        msg = "The excel file 'resources.xlsx', sheet 'classes' has a problem.\n"
+        msg += duplicate_check.execute_error_protocol()
+        raise InputError(msg)
+    # check that all the sheets have an entry in the names column and vice versa
+    if (all_names := set(classes_df["name"].tolist())) != (all_sheets := set(df_dict.keys())):
+        msg = ResourcesExcelSheetsNotAsExpected(all_names, all_sheets).execute_error_protocol()
+        raise InputError(msg)
```

Comment:
how would you feel about returning the Problem and then raising the error on the caller side?